src/main/cpp/src/hive_hash.cu

```diff
@@ -434,10 +470,17 @@ class hive_device_row_hasher {
 
    private:
     hash_functor_t const hash_functor;
+    cudf::size_type const _col_idx;
+    cudf::column_device_view* _flattened_column_views;
+    cudf::size_type* _first_child_index;
+    cudf::size_type* _nested_column_map;
```

Comment:
Don't need to store these pointers again. That will be very costly as they will occupy a lot of registers. Instead, just store one pointer to the parent class `hive_device_row_hasher`. The value `col_idx` can be passed in as function parameter instead.
```suggestion
  hive_device_row_hasher* parent;
```
Then:
```
__device__ hive_hash_value_t operator()(cudf::column_device_view const& col,
                                            cudf::size_type col_idx,
                                            cudf::size_type row_idx) const noexcept
{
  ...
  // Access column through `parent->_flattened_column_views[col_idx]`
  // Similar for the other variables.
}
```